### DISCUSSION TITLE
Error axios post request when deploying nextjs to vercel-help me

### DISCUSSION THREAD

**Discussion by DinhThuongNguyen** [vercel/next.js#23426] [Help]
code api login
`export default async function login (req, res){
  const {method} = req;
  if (method !== "POST") {
    return res.status(500).json({ message: "Request HTTP Method Incorrect." });
  } else {
    await validateBody(req, res);
  const errors = validationResult(req);
  if (!errors.isEmpty()) {
    return res.status(422).json({ errors: errors.array() });
  }
  const { email, password } = req.body;

  const checkEmail = await DBaccount.findOne({ email: email }).exec();
  const check_tenDangNhap = await DBaccount.findOne({ name: email }).exec();
  if (!checkEmail && !check_tenDangNhap) {
    return res.status(404).json({ message: "Tài khoản này không tồn tại" });
  }
  const result = checkEmail ? checkEmail : check_tenDangNhap;
  const checkPassword = await bcrypt.compare(password, result.password);

  if (!checkPassword) {
    return res.status(404).json({ message: "sai mật khẩu" });
  }

  let token;
  try {
    token = jwt.sign(
      {
        accountId: result.id,
        role: result.role,
      },
      process.env.EXAMPLE_TOKEN,
      { expiresIn: "8h" }
    );
  } catch (error) {
    return res.status(405).json({ message: "Create token failed" });
  }
  const options = {
    expires: new Date(Date.now() + JWT_EXPIRATION_NUM),
    secure: process.env.NODE_ENV === "production",
    httpOnly: process.env.NODE_ENV !== "production",
    sameSite: "lax",
    path: "/",
  };
  try {
    res.setHeader("Set-Cookie", cookie.serialize("vhy", token, options));
  } catch (error) {
    return res.status(405).json({ message: "Error set cookie" });
  }

  res.status(200).json({
    name: result.name,
    role: result.role,
    avatar: result.avatar,
    accountId: result.id,
  });
  }
}`

code pages login
`import React, { useContext, useEffect, useState } from "react";
import css from "./style.module.scss";
import BaseLayout from "../../components/Layouts/BaseLayout/BaseLayout";
import { Form, Formik } from "formik";
import {object, string} from "yup";
import FormikControls from "../../components/FormikForm/FormControls";
import Axios from "../../Axios/methodApi";
import { signIn } from 'next-auth/client';
import { AuthContext } from "../../ContextAPI/Auth-context";
import {useRouter} from "next/router";
import Link from "next/link";

const index = () => {
  const context = useContext(AuthContext);
  const [textError, setTextError] = useState();
  const route = useRouter();
  const initialValues = {
    email: "",
    password: "",
  };
  const validationSchema = object({
    email: string().required("Hãy nhập thông tin"),
    password: string().required("Chưa nhập mật khẩu"),
  });

  const onSubmit = (values) => {
    const dataLogin = {
      email: values.email,
      password: values.password
    };
    Axios.post("/account/login", JSON.stringify(dataLogin))
      .then(res => {
        context.login(
          res.accountId,
          res.role,
          res.name,
          res.avatar
        );
        setTextError("");
        route.push("/")
      })
      .catch(err => {
        console.log({err});
      })
  };
  return (
    <BaseLayout title="Đăng nhập tài khoản" >
      <div className={css.signup}>
        <div className={css.signup__content}>
          <h4>Đăng nhập</h4>
          <Formik
            initialValues={initialValues}
            validationSchema={validationSchema}
            onSubmit={onSubmit}
          >
            {(formik) => {
              return (
                <Form>
                  <FormikControls
                    control="input"
                    type="text"
                    label="Tên đăng nhập"
                    name="email"
                    placeholder="Nhập email hoặc tên tài khoản"
                  />
                  <FormikControls
                    control="input"
                    type="password"
                    label="Mật khẩu"
                    name="password"
                    placeholder="Nhập mật khẩu"
                  />
                    {textError && <p style={{color:"red", textAlign:"center", width: "100%"}}>{textError}</p>}
                    {textError === "sai mật khẩu" && <p style={{color:"black", textAlign:"center", width: "100%"}}>Nếu bạn quên mật khẩu, hãy <Link href="/quenmatkhau"><a style={{color: "blue"}}>tạo mật khẩu mới !</a></Link></p>}
                  <div className={css.btnSubmit}>
                    <button
                      type="submit"
                      disabled={!formik.isValid}
                      className={css.buttonControl}
                      className="mx-auto"
                    >
                      Đăng nhập
                    </button>
                  </div>
                  {formik.isSubmitting && (
                      <div className={css.loaded}>
                        <div className={css.sk__chase}>
                          <div className={css.sk__chase__dot}></div>
                          <div className={css.sk__chase__dot}></div>
                          <div className={css.sk__chase__dot}></div>
                          <div className={css.sk__chase__dot}></div>
                          <div className={css.sk__chase__dot}></div>
                          <div className={css.sk__chase__dot}></div>
                        </div>
                      </div>
                    )}
                </Form>
              );
            }}
          </Formik>
        </div>
      </div>
    </BaseLayout>
  );
};
export default index;
`
When I run the code on localhost, I can login, but when deploying to nextjs, I can't login

**Comment by mcsdevv** [comment]
Hi there, if you're having trouble with your code when deploying on Vercel please submit a [support request](https://vercel.com/support/request).

**Reply by LouieRichardson99** [reply]
I think I know why the API endpoint isn't working when deployed to Vercel.
I had this issue myself and It comes down to the bcrypt package.

Under the hood bcrypt uses C++ which is why it is leading to problems when deployed.
Try [bcryptjs](https://preview.npmjs.com/package/bcryptjs/v/1.0.1) as it is plain JavaScript.

Let me know how it goes.

**Comment by DinhThuongNguyen** [comment]
It is true that due to the bug related to bcrypt and I replaced it with
bcryptjs, it worked fine again.
Thank you man.

Vào CN, 18 thg 4, 2021 vào lúc 14:29 Louie Richardson <
***@***.***> đã viết:
